src/main/java/mate/jdbc/dao/impl/CarDaoImpl.java

```diff
@@ -0,0 +1,205 @@
+package mate.jdbc.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.jdbc.dao.CarDao;
+import mate.jdbc.dao.ManufacturerDao;
+import mate.jdbc.exception.DataProcessingException;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Car;
+import mate.jdbc.model.Driver;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    private static final int PARAMETER_INDEX_1 = 1;
+    private static final int PARAMETER_INDEX_2 = 2;
+    private static final int PARAMETER_INDEX_3 = 3;
+    private static final int COLUMN_INDEX_1 = 1;
+    private static final String ID_COLUMN = "id";
+    private static final String NAME_COLUMN = "name";
+    private static final String LICENSE_NUMBER_COLUMN = "license_number";
+    private static final String MODEL_COLUMN = "model";
+    private static final String MANUFACTURER_ID_COLUMN = "manufacturer_id";
+
+    @Override
+    public Car create(Car car) {
+        String query = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement statement = connection.prepareStatement(query,
+                         Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(PARAMETER_INDEX_1, car.getModel());
+            statement.setLong(PARAMETER_INDEX_2, car.getManufacturer().getId());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(COLUMN_INDEX_1, Long.class));
+            }
+            recordDrivers(car);
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String getCarQuery =
+                "SELECT id, model, manufacturer_id FROM"
+                       + " cars WHERE id = ? AND is_deleted = FALSE";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getCarStatement = connection
+                         .prepareStatement(getCarQuery)) {
+            getCarStatement.setLong(PARAMETER_INDEX_1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();;
+            if (resultSet.next()) {
+                car = getCarFromResultSet(resultSet);
+            }
+            if (car != null) {
+                car.setDrivers(getDriverById(id));
+            }
+            return Optional.ofNullable(car);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id " + id, e);
+        }
```

Comment:
```suggestion
            return Optional.ofNullable(car);
        } catch (SQLException e) {
            throw new DataProcessingException("Can't get car by id " + id, e);
        }
        if (car != null) {
                car.setDrivers(getDriverById(id));
        }
```
same, you should not open nested connections